pennylane/devices/task_qubit.py

```diff
@@ -0,0 +1,449 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module contains a proxy qubit object for spawning multiple instances of
+a given qubit type for run on a background scheduler.
+"""
+from typing import List, Union, Iterable
+
+from contextlib import nullcontext
+import pennylane as qml
+from pennylane import DeviceError
+
+from .default_qubit import DefaultQubit
+from .._version import __version__
+
+try:
+    import dask
+    from dask.distributed import worker_client, performance_report
+
+    # Ensure tasks are scheduled on processes rather than threads
+    dask.config.set(scheduler="processes")
+
+except ImportError as e:  # pragma: no cover
+    raise ImportError("task.qubit requires installing Dask.distributed") from e
+
+
+class ProxyHybridMethod:  # pragma: no cover
+    """
+    This utility class allows the use of both an instance
+    as well as class method types. For situations where
+    the explicit class method is needed, we can use this as a
+    decorator, and select the appropriate call dynamically.
+    This is an essential support for accessing backend
+    functionality supports with the proxy `task.qubit`.
+
+    The implementation is based on the Python descriptor guide
+    as mentioned at https://stackoverflow.com/questions/28237955/same-name-for-classmethod-and-instancemethod/28238047#28238047
+    """
+
+    def __init__(self, fclass, finstance=None, doc=None):
+        self.fclass = fclass
+        self.finstance = finstance
+        self.__doc__ = doc or fclass.__doc__
+        self.__isabstractmethod__ = bool(getattr(fclass, "__isabstractmethod__", False))
+
+    def classmethod(self, fclass):
+        """
+        Defines the decorated method as a class-method.
+        """
+        return type(self)(fclass, self.finstance, None)
+
+    def instancemethod(self, finstance):
+        """
+        Defines the decorated method as an instance-method.
+        """
+        return type(self)(self.fclass, finstance, self.__doc__)
+
+    def __get__(self, instance, cls):  # pragma: no cover
+        """
+        This allows the return of class variables and supported methods to be determined by the chosen backend, or native to the `task.qubit` class itself if not an instance.
+        """
+        if instance is None or self.finstance is None:
+            return self.fclass.__get__(cls, None)
+        return self.finstance.__get__(instance, cls)
+
+
+class TaskQubit(DefaultQubit):
+    """Proxy simulator plugin written using Dask.Distributed as a task-distribution scheduling backend.
+
+    **Short name:** ``task.qubit``
+
+    This device provides a pure-state qubit simulator wrapping both ``default.qubit`` and ``lightning.qubit``,
+    and written to allow batched offloading to a Dask scheduler. The ``task.qubit`` device works with the Autograd,
+    TensorFlow, PyTorch and (non-JIT) JAX interfaces. Currently, support exists for both parameter-shift and
```

Comment:
How about finite-diff?

Reply:
Good question. I never prioritised FD, but I think it should also work. Will attempt a trial.

Reply:
Just gave it a try. It seems working without any problem. I will add tests and update the description.